Library/Formula/berkeley-db.rb

```diff
@@ -33,6 +33,26 @@ def install
       --enable-cxx
       --enable-compat185
     ]
+    if build.with? "java"
+        java_home = ""
+        if ENV.has_key?("JAVA_HOME")
+            java_home = ENV["JAVA_HOME"]
+       else
+           java_home = `/usr/libexec/java_home`.chomp
```

Comment:
I don't think that is correct. JAVA_HOME determines that JDK to use. /usr/libexec/java_home returns the last in the list. So for example:

matthew@matthew-osx /u/l/L/Formula> echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/jdk1.7.0_60.jdk/Contents/Home
matthew@matthew-osx /u/l/L/Formula> java -version
java version "1.7.0_60"
Java(TM) SE Runtime Environment (build 1.7.0_60-b19)
Java HotSpot(TM) 64-Bit Server VM (build 24.60-b09, mixed mode)
matthew@matthew-osx /u/l/L/Formula> /usr/libexec/java_home 
/Library/Java/JavaVirtualMachines/jdk1.8.0_31.jdk/Contents/Home